watertap/core/membrane_channel_base.py

```diff
@@ -0,0 +1,904 @@
+            has_pressure_change=has_pressure_change, custom_term=custom_term
+        )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.length_domain,
+            doc="Pressure at interface",
+        )
+        def eq_equal_pressure_interface(b, t, x):
+            return b.properties_interface[t, x].pressure == b.properties[t, x].pressure
+
+        if has_pressure_change:
+            self._add_pressure_change(pressure_change_type=pressure_change_type)
+
+        if pressure_change_type == PressureChangeType.calculated:
+            self._add_calculated_pressure_change()
+
+    def add_isothermal_conditions(self):
+
+        # ==========================================================================
+        # Bulk and interface connections on the feed-side
+        # TEMPERATURE
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.length_domain,
+            doc="Temperature at interface",
+        )
+        def eq_equal_temp_interface(b, t, x):
+            return (
+                b.properties[t, x].temperature
+                == b.properties_interface[t, x].temperature
+            )
+
+    def add_extensive_flow_to_interface(self):
+        # VOLUMETRIC FLOWRATE
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.length_domain,
+            doc="Volumetric flow at interface of inlet",
+        )
+        def eq_equal_flow_vol_interface(b, t, x):
+            return (
+                b.properties_interface[t, x].flow_vol_phase["Liq"]
+                == b.properties[t, x].flow_vol_phase["Liq"]
+            )
+
+    def add_concentration_polarization(
+        self,
+        concentration_polarization_type=ConcentrationPolarizationType.calculated,
+        mass_transfer_coefficient=MassTransferCoefficient.calculated,
+    ):
+
+        solute_set = self.config.property_package.solute_set
+        units_meta = self.config.property_package.get_metadata().get_derived_units
+
+        if concentration_polarization_type == ConcentrationPolarizationType.none:
+
+            @self.Constraint(
+                self.flowsheet().config.time,
+                self.length_domain,
+                solute_set,
+                doc="Concentration polarization",
+            )
+            def eq_concentration_polarization(b, t, x, j):
+                return (
+                    b.properties_interface[t, x].conc_mass_phase_comp["Liq", j]
+                    == b.properties[t, x].conc_mass_phase_comp["Liq", j]
+                )
+
+            return
+
+        if concentration_polarization_type not in (
+            ConcentrationPolarizationType.fixed,
+            ConcentrationPolarizationType.calculated,
+        ):
+            raise ConfigurationError(
+                f"Unrecognized concentration_polarization_type {concentration_polarization_type}"
+            )
+
+        self.cp_modulus = Var(
```

Comment:
should there be an elif above this, basically checking that CP type is not none? If not none, then create cp_modulus